src/c-meta.rkt

```diff
@@ -125,20 +125,50 @@
             (for/list ([decl (decl:vars-declarators stmt)])
               (let* ([init (decl:declarator-initializer decl)]
                      [type (decl:declarator-type decl)])
-
                 ; Assumes this is float or float array
-                (define initializer
-                  (cond
-                    [init
-                      (translate (init:expr-expr init))]
-                    [(type:array? type)
-                      `(make-v-list-zeros (unquote (multi-array-length type)))]
-                    [(type:primitive? type) 0]
-                    [else (error "unexpected initializer in declaration" stmt)]))
-                (quasiquote
-                  (define
-                    (unquote (translate (decl:declarator-id decl)))
-                    (unquote initializer))))))
+                (if (init:compound? init)
+                  (begin 
+                    (define arr-name (translate (decl:declarator-id decl)))
+                      (define (make-quasi-list init-list counter acc)
+                        (if (null? init-list) 
+                            acc
+                            (make-quasi-list 
+                              (cdr init-list) 
+                              (- counter 1)
+                              (cons 
+                                (let ([inner_expr (init:expr-expr (car init-list))])
+                                  (quasiquote
+                                    (v-list-set! 
+                                    (unquote arr-name)
+                                    (unquote counter)
+                                    (unquote 
+                                      (cond 
+                                        [(id:var? inner_expr) 
```

Comment:
This looks great! 

One suggestion that is a little more Racket-ey and might improve readability: instead of recurring on each element of the list with a counter and using `v-list-set!`, an alternative would be to map the translation over the values first, then convert them to a `v-list` in a second step. You can do this using this function from `utils.h` which takes a variadic number of arguments:

```lang=racket
(define (v-list . xs)
  # ...
)
```

In Racket, you can pass a list of values to a function that expects variadic arguments using the [apply](https://docs.racket-lang.org/reference/procedures.html#%28def._%28%28lib._racket%2Fprivate%2Fbase..rkt%29._apply%29%29) function. 

The is, if we have some list `mapped-values`, we can make a v-list with:

```lang=racket
(apply v-list mapped-values)
```

That way, you can avoid using an explicit counter - does that make sense?

Reply:
I've changed the code to use 2 maps, the only issue appears to be that it does not generate the `PDX_Z_ALIGN();` function. Otherwise there are no differences than what was generated before.

Reply:
To avoid committing the changes to the `PDX_Z_ALIGN();` lines in the other files, could you merge master into this branch, rerun the tests on this branch, then add a new commit? Then this PR will only cover the files you intend to change. Thanks!

Reply:
Yep! I've made the changes for the tests back to what they were, reran and checked they still pass, and I've added it in the newest commit.